### PR TITLE
[JS] feat: Adaptive Cards authentication 

### PR DESCRIPTION
**Pull request by singhk97**
## Linked issues

closes: #817

## Details

Added support for user authentication using Adaptive Cards Universal Actions (Action.Execute)

#### Change details

- [x] Added a new `AdaptiveCardAuthentication` class that handles all adaptive card auth related scenarios. 
- [x] Created `OAuthSettings` that is a superset of `OAuthPromptSettings` to include the `tokenExchangeUri` property needed for Adaptive Cards SSO. 
- [x] Added a sample to show case adaptive cards
- [ ] Added unit tests
- [x] Updated sample docs

> Describe your changes, with screenshots and code snippets as appropriate

**code snippets**:

**screenshots**:

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (we use [TypeDoc](https://typedoc.org/) to document our code)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

File: js/samples/06.auth.adaptiveCardSSO/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+// Import required bot services.
+// See https://aka.ms/bot-services to learn more about the different parts of a bot.
+import {
+    ActivityTypes,
+    CardFactory,
+    CloudAdapter,
+    ConfigurationBotFrameworkAuthentication,
+    ConfigurationBotFrameworkAuthenticationOptions,
+    MemoryStorage,
+    MessagingExtensionAttachment,
```
Comment (corinagum) [inline js/samples/06.auth.adaptiveCardSSO/src/index.ts:19]: I know you're going to hate this but unfortunately, `MessagingExtension(s)` is the incorrect terminology. `MessageExtension` is correct. Please do a rename (I asked the same for Chaoyi's PR). 

- if it's an import from botbuilder, it's fine to leave it as-is. We can't do anything about pre-existing incorrect naming.
- From here on out, any instance of message extension related names should be kept as MessageExtension, and Messaging... should be avoided as much as possible. 
- Same is applicable to samples

Sorry for not catching this earlier.

File: js/samples/06.auth.messagingExtensionSSO/teamsapp.local.yml [hunk 2]
```diff
@@ -16,15 +16,15 @@ provision:
   # Automates the creation of a Teams app registration and saves the App ID to an environment file.
   - uses: teamsApp/create
     with:
-      name: SsoAuthBot-${{TEAMSFX_ENV}}
+      name: MessagingExtensionSSO-${{TEAMSFX_ENV}}
```
Comment (corinagum) [inline js/samples/06.auth.messagingExtensionSSO/teamsapp.local.yml:19]: Please rename

File: js/packages/teams-ai/src/authentication/Authentication.ts [hunk 3]
```diff
@@ -14,38 +14,43 @@ import { Application, Selector } from '../Application';
 import { MessagingExtensionAuthentication } from './MessagingExtensionAuthentication';
```
Comment (corinagum) [inline js/packages/teams-ai/src/authentication/Authentication.ts:14]: I know you're going to hate this but unfortunately, `MessagingExtension(s)` is the incorrect terminology. `MessageExtension` is correct. Please do a rename (I asked the same for Chaoyi's PR). 

- if it's an import from botbuilder, it's fine to leave it as-is. We can't do anything about pre-existing incorrect naming.
- From here on out, any instance of message extension related names should be kept as MessageExtension, and Messaging... should be avoided as much as possible. 
- Same is applicable to samples

Sorry for not catching this earlier.

File: js/samples/06.auth.adaptiveCardSSO/README.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
+# Teams Adaptive Card SSO Bot
+
+This sample shows how to incorporate a basic conversational flow into a Microsoft Teams application using [Bot Framework](https://dev.botframework.com) and the Teams AI SDK.
+
```
Comment (corinagum) [inline js/samples/06.auth.adaptiveCardSSO/README.md:4]: Fix

File: js/samples/06.auth.messagingExtensionSSO/README.md [hunk 5]
```diff
@@ -90,7 +90,7 @@ You can also use the Teams Toolkit CLI to run this sample.
 1. In the repository directory, run the Teams Toolkit CLI commands to automate the setup needed for the app
 
     ```bash
-    cd teams-ai/js/samples/01.messaging.a.SsoAuthBot/
+    cd teams-ai/js/samples/01.messaging.a.echoBot/
```
Comment (corinagum) [inline js/samples/06.auth.messagingExtensionSSO/README.md:93]: path is incorrect

File: js/samples/06.auth.adaptiveCardSSO/src/graphClient.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+            }
+        });
+    }
+
+    public async GetMyProfile() {
+        return await this.graphClient.api('/me').get();
+    }
+
+    // Gets the user's photo
+    public async GetPhotoAsync(): Promise<string> {
```
Comment (corinagum) [inline js/samples/06.auth.adaptiveCardSSO/src/graphClient.ts:35]: Should this be `getProfilePhotoAsync` per MS graph API?

File: js/samples/06.auth.adaptiveCardSSO/src/graphClient.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+        // Get an authenticated Microsoft Graph client using the token issued to the user.
+        this.graphClient = Client.init({
+            authProvider: (done) => {
+                done(null, this._token); // First parameter takes an error if you can't get an access token.
+            }
+        });
+    }
+
+    public async GetMyProfile() {
```
Comment (corinagum) [inline js/samples/06.auth.adaptiveCardSSO/src/graphClient.ts:30]: pascal case

File: js/packages/teams-ai/src/authentication/Authentication.ts [hunk 8]
```diff
@@ -245,6 +254,16 @@ export class AuthenticationManager<TState extends TurnState> {
     }
 }
 
+/**
+ * Settings used to configure user authentication through the OAuthPrompt.
+ */
+export type OAuthSettings = OAuthPromptSettings & {
```
Comment (corinagum) [inline js/packages/teams-ai/src/authentication/Authentication.ts:260]: Please check with Chaoyi & align on how to do casing for OAuth (oAuth, oauth, etc)

**Review by corinagum — CHANGES_REQUESTED** [review]
Please see comments, and a quick check:
- pascal casing except for class names and typings
- note about MessageExtension naming - please sync with Chaoyi as needed.

File: js/samples/06.auth.messagingExtensionSSO/teamsapp.local.yml (hunk 2 above)
Comment (singhk97) [reply js/samples/06.auth.messagingExtensionSSO/teamsapp.local.yml:19]: done

File: js/packages/teams-ai/src/authentication/Authentication.ts (hunk 3 above)
Comment (singhk97) [reply js/packages/teams-ai/src/authentication/Authentication.ts:14]: done

File: js/samples/06.auth.adaptiveCardSSO/README.md (hunk 4 above)
Comment (singhk97) [reply js/samples/06.auth.adaptiveCardSSO/README.md:4]: https://github.com/orgs/microsoft/projects/719?pane=issue&itemId=38284723

File: js/samples/06.auth.messagingExtensionSSO/README.md (hunk 5 above)
Comment (singhk97) [reply js/samples/06.auth.messagingExtensionSSO/README.md:93]: fixed

File: js/samples/06.auth.adaptiveCardSSO/src/graphClient.ts (hunk 6 above)
Comment (singhk97) [reply js/samples/06.auth.adaptiveCardSSO/src/graphClient.ts:35]: fixed

File: js/samples/06.auth.adaptiveCardSSO/src/graphClient.ts (hunk 7 above)
Comment (singhk97) [reply js/samples/06.auth.adaptiveCardSSO/src/graphClient.ts:30]: fixed

File: js/packages/teams-ai/src/authentication/Authentication.ts (hunk 8 above)
Comment (singhk97) [reply js/packages/teams-ai/src/authentication/Authentication.ts:260]: Ok

**Review by corinagum — APPROVED** [review]

File: js/packages/teams-ai/src/authentication/Authentication.ts (hunk 8 above)
Comment (blackchoey) [reply js/packages/teams-ai/src/authentication/Authentication.ts:260]: `OAuth` looks good to me, as it's used in the RFC spec. For variables that need to start with lowercase based on the naming rule, we may use `oAuth` in my opinion. @singhk97 @corinagum

**Pull request merged**